version: "3.3"
   
services:
    django:
        build: .
        container_name: django
        command: python manage.py runserver 0.0.0.0:8000
        volumes:
            - .:/usr/src/app/
        ports:
            - "7000:8000"
        environment:
            - DEBUG=1
            - CELERY_BROKER_URL=redis://redis:6379/0
            - CELERY_RESULT_BACKEND=redis://redis:6379/0
            - DATABASE_URL=postgres://postgres:randomized@cloud_sql_proxy:5432/postgres
            - REDIS_CHANNEL_HOST=redis
            - REDIS_CHANNEL_PORT=6379
        depends_on:
            - cloud-sql-proxy
            - redis
    celery:
        build: .
        command: celery -A config.celery_app worker -l INFO
        volumes:
            - .:/usr/src/app
        environment:
            - DEBUG=1
            - CELERY_BROKER_URL=redis://redis:6379/0
            - CELERY_RESULT_BACKEND=redis://redis:6379/0
            - DATABASE_URL=postgres://postgres:randomized@cloud_sql_proxy:5432/postgres

        depends_on:
            - django
            - redis

    cloud-sql-proxy:
        image: gcr.io/cloudsql-docker/gce-proxy:1.11
        # Port: MySQL (3306), PostgreSQL (5433)
        command: /cloud_sql_proxy -instances=main-291913:europe-west2:fileshare=tcp:0.0.0.0:5474 -credential_file=/config
        volumes:
          - ./credentials.json:/config
        ports:
          - 5474:5474


    redis:
        image: "redis:alpine"
        
# volumes:
#     pgdata: